### DISCUSSION TITLE
Settings management examples

### DISCUSSION THREAD

**Discussion by thmsklngr** [pydantic/pydantic#3582] [Question] [answered]
Hi there,
I just read an interesting Medium article describing Pydantic's ability for handling configuration settings, which is really of high interest for me.
I followed the examples from the docs and get `value_error.missing`, even though I set them either through `os.environ` or using `export VARIABLE`. But not for all variables, not for all. Example:
```bash
export V0=0
export SUB_MODEL='{"v1": "json-1", "v2": "json-2"}'
export SUB_MODEL__V2=nested-2
export SUB_MODEL__V3=3
export SUB_MODEL__DEEP__V4=v4
```
```python
from pydantic import BaseModel, BaseSettings

class DeepSubModel(BaseModel):
    v4: str

class SubModel(BaseModel):
    v1: str
    v2: bytes
    v3: int
    deep: DeepSubModel

class Settings(BaseSettings):
    v0: str
    sub_model: SubModel

    class Config:
        env_nested_delimiter = '__'
```
As a result running `print(Settings().dict())` I get an error instead:
```
pydantic.error_wrappers.ValidationError: 2 validation errors for Settings
sub_model -> v3
  field required (type=value_error.missing)
sub_model -> deep
  field required (type=value_error.missing)
```
Strange to me is that I only get errors for v3 and deep, but not for others. This kind of behavior happens to me also when trying to run example #1 at the top of the docs page. Either I'm missing something or the docs are somehow out of date.
System information:
```
             pydantic version: 1.8.2
            pydantic compiled: False
                 install path: /Users/thomas/Projekte/Entwicklung/Python/pydantic/venv/lib/python3.10/site-packages/pydantic
               python version: 3.10.1 (main, Dec  6 2021, 23:19:43) [Clang 12.0.0 (clang-1200.0.32.29)]
                     platform: macOS-10.15.7-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions']
```
Any ideas?

Regards, Thomas

**Comment by thmsklngr** [comment]
Ok, finding: it seems that whenever I set `sub_model` through `export SUBMODEL=`it works, but Pydantic ignored those settings set with the delimiter. So, in the example v3 and deep are set through `SUB_MODEL__V3` (and so on) are not covered.

**Comment by samuelcolvin** [comment, marked as answer]
The docs are for the pre-release version of pydantic v1. 9.0a2. you have version v1.8.2 installed.

For you to fix:
* Use the old docs through archive.org
* Or, install the pre-release version of pydantic

For us to do (issue welcome to track):
* Not deploy the docs for pre-release versions
* Keep a history of docs for old versions

**Reply by samuelcolvin** [reply]
See #3159

**Reply by thmsklngr** [reply]
Thanks for your reply. That makes sense, indeed. I did not recognize the version difference for the documentation, I was jumping directly into the docs without reading the front page.

**Reply by thmsklngr** [reply]
For those who are looking for the docs of the current stable release, here they are:
https://web.archive.org/web/20211217201708/https://pydantic-docs.helpmanual.io/
